xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/CompareVersionsTest.java

```diff
@@ -276,6 +279,64 @@ public void testUnifiedDiffShowsInlineChanges()
             "+alice = <ins>2 * </ins>bob;", " bob = tmp;");
     }
 
+    /**
+     * Tests navigation between versions.
+     */
+    private void testVersionNavigation()
+    {
+        String queryString = "viewer=changes&rev1=1.2&rev2=6.4";
+        getUtil().gotoPage(getTestClassName(), testPage.getMetaDataValue("page"), "view", queryString);
+        ChangesPane changesPane = new ChangesPane();
+
+        assertTrue(changesPane.getFromVersionSummary().startsWith("From version < 1.2 >"));
+        assertTrue(changesPane.getToVersionSummary().startsWith("To version < 6.4 >"));
+
+        By arrowLeftShiftSelector = By.cssSelector("#changes-info-boxes > a.changes-arrow-left");
```

Comment:
This is still exposing/leaking implementation details of the UI into the test.

As an example: If the UI is slightly updated (i.e. class name changed) and you have 5 tests for that UI, you will need to update the UI + 5 tests. If, instead, you encapsulate the UI's implementation details inside the page object class, when the UI changes, you only update the page object and all the tests that are using it will not need to be changed.

You have added the methods to the `ChangesPage` page object in order to click on the links. If you need to check in your tests that you *can* click on those links, i.e. if they exist, then you can extend the page object even more and add methods like:
* hasPreviousFromVersion()
* hasNextFromVersion()
* hasPreviousToVersion()
* hasNextToVersion()
* hasPreviousChange()
* hasNextChange()

Also note when designing the page object that you want to expose the functionality of the UI and not its implementation. E.g., a better name for `clickArrowLeftShift` would be `clickPreviousChange` and instead of `clickArrowLeftFrom` would be `clickPreviousFromVersion`, in the sense that the `arrowLeft` is an implementation detail that might change at a future redesign, while the functionality of "previousChange"/"previousVersion" will continue to exist in the UI.